### PR TITLE
PAYARA-1132 Make Enabling Slow SQL Logging Dynamic

### PR DESCRIPTION
**Pull request by jGauravGupta**

### CODE REVIEW

**Comment by jGauravGupta** [comment]
jenkins test please

**Comment by Cousjava** [comment]
jenkins test please

File: appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/spi/ManagedConnectionImpl.java [hunk 1]
```diff
@@ -1320,4 +1335,20 @@ public void purgeStatementFromCache(PreparedStatement preparedStatement) {
...
+         ManagedConnectionFactoryImpl spiMCF = (ManagedConnectionFactoryImpl) mcf;
+        Resources resources = Globals.getDefaultHabitat().getService(Domain.class).getResources();
+        ResourcePool pool = (ResourcePool) ConnectorsUtil.getResourceByName(resources, ResourcePool.class, spiMCF.getPoolName());
+        if(pool instanceof JdbcConnectionPool){
+            jdbcConnectionPool = (JdbcConnectionPool)pool;
+        }
+        return jdbcConnectionPool;
+    }
+
+    private boolean isSlowQueryLoggingEnable(){
```
Comment (mulderbaba) [inline appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/spi/ManagedConnectionImpl.java]: better to have the method name suffixed as Enabled?

File: appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/spi/ManagedConnectionImpl.java [hunk 2]
```diff
@@ -483,42 +493,47 @@ public Object getConnection(Subject sub, javax.resource.spi.ConnectionRequestInf
...
-                if (!sqlTraceDelegator.listenersRegistered()) {
-                    sqlTraceDelegator = null;
+                sqlTraceDelegator.deregisterSQLTraceListener(RequestTracingListener.class);
+            }
+
+            sqlTraceDelegator.deregisterSQLTraceListener(SlowSQLLogger.class);
+            if (connectionPool != null && isSlowQueryLoggingEnable()) {
+                double threshold = Double.valueOf(connectionPool.getSlowQueryThresholdInSeconds());
+                if (threshold > 0) {
+                    sqlTraceDelegator.registerSQLTraceListener(new SlowSQLLogger((long) (threshold * 1000)));
```
Comment (mulderbaba) [inline appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/spi/ManagedConnectionImpl.java]: using TimeUnit inside constructor parameter would make this more readable

**Pull request closed without merging**